src/libraries/System.Collections/ref/System.Collections.cs

```diff
@@ -47,6 +47,8 @@ public static partial class CollectionExtensions
         public static TValue GetValueOrDefault<TKey, TValue>(this System.Collections.Generic.IReadOnlyDictionary<TKey, TValue> dictionary, TKey key, TValue defaultValue) { throw null; }
         public static bool Remove<TKey, TValue>(this System.Collections.Generic.IDictionary<TKey, TValue> dictionary, TKey key, [System.Diagnostics.CodeAnalysis.MaybeNullWhenAttribute(false)] out TValue value) { throw null; }
         public static bool TryAdd<TKey, TValue>(this System.Collections.Generic.IDictionary<TKey, TValue> dictionary, TKey key, TValue value) { throw null; }
+        public static System.Collections.ObjectModel.ReadOnlyCollection<T> AsReadOnly<T>(this IList<T> list) { throw null; }
+        public static System.Collections.ObjectModel.ReadOnlyDictionary<TKey, TValue> AsReadOnly<TKey, TValue>(this IDictionary<TKey, TValue> dictionary) { throw null; }
```

Comment:
OK, so it looks like the issue stems from the fact that `S.C.O.ReadOnlyDictionary<TKey, TValue>` does not exist in System.Private.CoreLib but instead lives in the `System.ObjectModel` assembly (unlike `ReadOnlyCollection<T>`). I suspect this means that the particular API cannot be added unless we also move `ReadOnlyDictionary` to System.Private.CoreLib. I'm not sure how much of a breaking change such a move would be.

cc @jkotas @stephentoub

Reply:
If moving ReadOnlyCollection is not an option, would it be an option to add the extension method for IDictionary in the System.ObjectModel assembly? Maybe an ReadOnlyDictionaryExtensions class?

Reply:
It is fine to move it. We have done many moves like that.

Reply:
(I assume that it will move cleanly and that it won't bring some bad dependencies with it.)

Reply:
@jkotas I pushed f85ab090ac2cdf8b111b05a605f0c046f9764d87 which moves the type to S.P.C. Seems like a fairly straightforward transfer dependency-wise. The move triggered a couple of `MembersMustExist` errors in ApiCompat, added relevant suppressions.

Reply:
> The move triggered a couple of MembersMustExist errors in ApiCompat, added relevant suppressions.

Is it correct to suppress these? I think you may be missing a type forwarders from the original place.

Reply:
You mean [this declaration](https://github.com/dotnet/runtime/commit/f85ab090ac2cdf8b111b05a605f0c046f9764d87#diff-00bf31ba103c6bbb53786b76e20dada7a1cbf0404a3d0b9591027602a8b58fd9R13)? Or am I missing other configuration?

Reply:
I meant type forwards like https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Data.Common/src/System.Data.Common.TypeForwards.cs#L4 and https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/libraries/System.Data.Common/ref/System.Data.Common.Forwards.cs#L7 

The CI test failures are due to the missing type forwards too.

Reply:
Ok, so presumably I'd need to create a type forward in System.ObjectModel. I would have expected one to already exist for `ReadOnlyCollection<T>` but I couldn't find it.

Reply:
ReadOnlyCollection was in CoreLib/System.Runtime to start with.

Reply:
I added a `[assembly:TypeForwardedTo(typeof(ReadOnlyDictionary<,>)]` in both src and ref assemblies of `System.ObjectModel` but that didn't seem to remove the need for `ApiCompat` suppressions.

Reply:
@jkotas the build is now green but it seems like the suppressions were still necessary.

Reply:
ok